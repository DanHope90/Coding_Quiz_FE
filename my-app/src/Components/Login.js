/* eslint-disable no-unused-vars */
import React, { useState } from "react";
import axios from "axios";
import Alert from "./Alert";
import "../Styles/Register.css";

function Login() {
  const [loginData, setLoginData] = useState({
    email: "",
    password: "",
  });

  const { email, password } = loginData;

  const alertState = {
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const [alert, setAlert] = useState(alertState.alert);

  const [success, setSuccess] = useState(false);

  const handleSubmit = async (event) => {
    console.log({ ...loginData });
    event.preventDefault();
    setAlert({ message: "", isSuccess: false });
    axios
      .post(`http://localhost:4000/api/user/login`, { ...loginData })
      .then(() => {
        setAlert({
          message: "You have successfully logged in!",
          isSuccess: true,
        });
      })
      .catch((err) => {
        setAlert({
          message: "Login failed.",
          isSuccess: false,
        });
      });
  };

  function loginHandleChange(event) {
    setLoginData((prevState) => ({
      ...prevState,
      [event.target.name]: event.target.value,
    }));
  }

  return (
    <div className="titles">
      {alert.isSuccess && alert.message
        ? (
          <Alert
            message={alert.message}
            isSuccess={alert.isSuccess}
          />
        )
        : (
          <div>
            <Alert message={alert.message} isSuccess={alert.isSuccess} />
            <h1 className="titles">Login</h1>
            <h2 className="titles">Please submit your username and password</h2>
            <div>
              <form className="form" onSubmit={handleSubmit}>
                <input
                  type="email"
                  className="form-control"
                  id="email"
                  name="email"
                  value={email}
                  placeholder="Enter email address"
                  onChange={loginHandleChange}
                  required
                />
                <input
                  type="password"
                  className="form-control"
                  id="password"
                  name="password"
                  value={password}
                  placeholder="Enter password"
                  onChange={loginHandleChange}
                  required
                />
                <div className="register-button">
                  <button
                    type="submit"
                    className="button"
                  >
                    Login
                  </button>
                </div>
              </form>
            </div>
          </div>
        )}
    </div>

  );
}

export default Login;
