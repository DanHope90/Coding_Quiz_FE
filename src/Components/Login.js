/* eslint-disable no-underscore-dangle */
import React, { useState, useContext } from "react";
import axios from "axios";
import Context from "../Utils/Context";
import Alert from "./Alert";
import "../Styles/Register.css";

function Login() {
  const [loginData, setLoginData] = useState({
    email: "",
    password: "",
  });

  const { email, password } = loginData;

  const alertState = {
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const { setIsLoggedIn, setUserInfo } = useContext(Context);

  const [alert, setAlert] = useState(alertState.alert);

  const handleSubmit = async (event) => {
    event.preventDefault();
    setAlert({ message: "", isSuccess: false });
    axios
      .post(`https://dev-quiz-22.herokuapp.com/api/user/login`, { ...loginData })
      .then((res) => {
        setIsLoggedIn(true);
        setUserInfo({
          id: res.data._id,
          userName: res.data.userName,
          email: res.data.email,
          completedQuizzes: res.data.savedQuizzes,
        });
      })
      .catch(() => {
        setAlert({
          message: "Login failed.",
          isSuccess: false,
        });
      });
  };

  function loginHandleChange(event) {
    setLoginData((prevState) => ({
      ...prevState,
      [event.target.name]: event.target.value,
    }));
  }

  return (
    <div className="titles">
      <div>
        <Alert message={alert.message} isSuccess={alert.isSuccess} />
        <div className="form-body">
          <h1 className="titles">Welcome!</h1>
          <h2 className="titles">Use the bar above to take a quiz.</h2>
          <h2 className="titles">If you login, you can save/view your scores.</h2>
          <h2 className="titles">To login, submit your email and password below.</h2>
        </div>
        <div>
          <form className="form" onSubmit={handleSubmit}>
            <input
              type="email"
              className="form-control"
              id="email"
              name="email"
              value={email}
              placeholder="Enter email address"
              onChange={loginHandleChange}
              required
            />
            <input
              type="password"
              className="form-control"
              id="password"
              name="password"
              value={password}
              placeholder="Enter password"
              onChange={loginHandleChange}
              required
            />
            <div>
              <button
                type="submit"
                className="register-button"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>

  );
}

export default Login;
