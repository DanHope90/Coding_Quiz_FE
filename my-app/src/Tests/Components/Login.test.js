import React from "react";
import renderer from "react-test-renderer";
import Login from "../../Components/Login";

test("renders the Login component correctly", () => {
  const testRenderer = renderer.create(<Login />);
  expect(testRenderer).toMatchSnapshot();
});
