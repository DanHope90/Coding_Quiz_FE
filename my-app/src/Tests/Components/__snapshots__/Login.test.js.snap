// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Login component correctly 1`] = `
<div
  className="titles"
>
  <div>
    <h1
      className="titles"
    >
      Login
    </h1>
    <h2
      className="titles"
    >
      Please submit your username and password
    </h2>
    <div>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Enter email address"
          required={true}
          type="email"
          value=""
        />
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          required={true}
          type="password"
          value=""
        />
        <div
          className="userLogin"
        >
          <button
            className="button"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
